### PR TITLE
disallow some keywords to be specified in both salus.yaml and semgrep…

### PR DESCRIPTION
**Pull request by ghbren**
Disallow `pattern`,  `language`, and `message` for the same rule to be defined in both salus.yaml and semgrep config.
Values in semgrep config currently overrides salus.yaml, and may cause confusions to the user.

### CODE REVIEW

File: lib/salus/scanners/semgrep.rb [hunk 1]
```diff
@@ -39,6 +39,19 @@ def run
           # semgrep has the following behavior:
           #   ouputs a json object with a 'results' and 'errors' key
 
+          if !match['config'].nil?
+            override_keys = %w[pattern language message]
```
Comment (cbtoni) [inline lib/salus/scanners/semgrep.rb:43]: Is it `language` or `languages`? It also looks like there is both `pattern` and `patterns` allowed in the semgrep config 🤔

File: lib/salus/scanners/semgrep.rb [hunk 2]
```diff
@@ -39,6 +39,19 @@ def run
...
+            override_keys.any? do |k|
```
Comment (cbtoni) [inline lib/salus/scanners/semgrep.rb:44]: I think you mean `override_keys.each` here?

Comment (ghbren) [reply lib/salus/scanners/semgrep.rb:44]: yes, thank you

File: lib/salus/scanners/semgrep.rb (hunk 1 above)
Comment (ghbren) [reply lib/salus/scanners/semgrep.rb:43]: * salus.yaml only supports `language`, semgrep yaml only supports `languages`. 
  - semgrep yaml requires `languages` and will override `language` from salus.yaml
* salus.yaml only supports `pattern`, semgrep yaml supports `pattern` and `patterns`
  - semgrep yaml requires either `pattern` or `patterns`, and will override `pattern` from salus.yaml

File: lib/salus/scanners/semgrep.rb (hunk 2 above)
Comment (ghbren) [reply lib/salus/scanners/semgrep.rb]: @cbtoni I just addressed the comments and updated error message.  Could you re-review? Thanks

**Pull request merged**